components/range_cache_memory_engine/src/range_manager.rs

```diff
@@ -44,22 +68,56 @@ impl SnapshotList {
     }
 }
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub struct RangeMeta {
-    // start_key and end_key cannot uniquely identify a range as range can split and merge, so we
-    // need a range id.
-    id: u64,
-    range_snapshot_list: SnapshotList,
+    region: Region,
+    range: CacheRange,
+    region_snapshot_list: SnapshotList,
     safe_point: u64,
+    state: RegionState,
+    in_gc: bool,
+    evict_info: Option<EvictInfo>,
+}
+
+#[derive(Debug, Clone, Copy)]
+struct EvictInfo {
+    start: Instant,
+    reason: EvictReason,
 }
 
 impl RangeMeta {
-    fn new(id: u64) -> Self {
+    fn new(region: Region) -> Self {
+        let range = CacheRange::from_region(&region);
         Self {
-            id,
-            range_snapshot_list: SnapshotList::default(),
+            region,
+            range,
+            region_snapshot_list: SnapshotList::default(),
             safe_point: 0,
+            state: RegionState::Pending,
+            in_gc: false,
+            evict_info: None,
+        }
+    }
+
+    #[inline]
+    pub(crate) fn region(&self) -> &Region {
+        &self.region
+    }
+
+    // update region info due to epoch changes.
+    pub(crate) fn update_region(&mut self, region: &Region) -> bool {
+        assert!(
+            self.region.id == region.id
+                && self.region.get_region_epoch().version < region.get_region_epoch().version
+        );
+        let new_range = CacheRange::from_region(region);
+        if !self.range.contains_range(&new_range) {
```

Comment:
So the `update_region` is just for split or epoch update?

Reply:
Only used for outdated pending regions that missing some newer epoch changes events. For region split, we use remove & add to replace the old region with new split regions.